NGitLab.Mock/Clients/ProjectIssueNoteClient.cs

```diff
@@ -14,24 +16,59 @@ public ProjectIssueNoteClient(ClientContext context, int projectId)
             _projectId = projectId;
         }
 
-        public ProjectIssueNote Create(ProjectIssueNoteCreate create)
+        public Models.ProjectIssueNote Create(ProjectIssueNoteCreate create)
         {
-            throw new NotImplementedException();
+            using (Context.BeginOperationScope())
+            {
+                var project = GetProject(_projectId, ProjectPermission.Contribute);
+                var issue = project.Issues.First(iss => iss.Id == create.IssueId);
```

Comment:
I think you should use FirstOrDefault and throw `GitLabNotFoundException` if the value is null. This way it simulates returns a 404 error instead of throwing a _random_ exception.

Also, you can do similar changes in all other methods